packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/exposurepoint/ExposurePointFeature.java

```diff
@@ -78,9 +79,13 @@ public void updateBuilder(@NonNull CaptureRequest.Builder requestBuilder) {
     if (!checkIsSupported()) {
       return;
     }
-    requestBuilder.set(
-        CaptureRequest.CONTROL_AE_REGIONS,
-        exposureRectangle == null ? null : new MeteringRectangle[] {exposureRectangle});
+    if (exposureRectangle != null) {
+      requestBuilder.set(
+          CaptureRequest.CONTROL_AE_REGIONS, new MeteringRectangle[] {exposureRectangle});
+    } else if (shouldReset(requestBuilder)) {
+      defaultExposureRectangle = requestBuilder.get(CaptureRequest.CONTROL_AE_REGIONS);
+      requestBuilder.set(CaptureRequest.CONTROL_AE_REGIONS, defaultExposureRectangle);
+    }
```

Comment:
To follow up on https://github.com/flutter/packages/pull/3718#discussion_r1169379805, I understand what you're saying, but where I'm unclear is the case where `exposureRectangle` was previous non-null (so ` CaptureRequest.CONTROL_AE_REGIONS` was set), but has now been changed to null, but we want to call `requestBuilder.set(CaptureRequest.CONTROL_AE_REGIONS, null)` as we did before.

With this code change, it seems to be that we will not do this because `shouldReset(...)` will return false.

Reply:
You are correct that the previous implementation would set the CONTROL_AE_REGIONS to null even if exposureRectangle was previously non-null. I apologize for my mistake in not considering this scenario. I have updated the code to ensure that CONTROL_AE_REGIONS is only set to null when exposureRectangle is null and shouldReset() returns true.
You are correct that the previous implementation would set the CONTROL_AE_REGIONS to null even if exposureRectangle was previously non-null. I apologize for my mistake in not considering this scenario. I have updated the code to ensure that CONTROL_AE_REGIONS is only set to null when exposureRectangle is null and shouldReset() returns true.

Reply:
`exposureRectangle` would still be null in that case, so wouldn't this problem still be here? Also, I'm not sure we ever expect `requestBuilder` to be null.

One solution could be using a boolean to determine whether or not `exposureRectangle` was previously set to a non-null value and based on that, determining whether or not you should fall into the first case versus checking if `exposureRectangle` is null. That way, `CaptureRequest.CONTROL_AE_REGIONS` will not be set to `null` by default, but can be set to that if so desired.

Reply:
I made an improvement on shouldReset to meet the solution